docs: Adds guide for customizing page titles with TitleStrategy

SkyZeroZx opened the pull request:
Improves documentation by explaining how to implement and configure a custom title strategy for centralized control over page titles

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on adev/src/content/guide/routing/define-routes.md:
```diff
@@ -299,6 +299,48 @@ const routes: Routes = [
+@Injectable({
+    providedIn: 'root',
+})
+export class AppTitleStrategy extends DefaultTitleStrategy {
```
```suggestion
export class AppTitleStrategy extends TitleStrategy {
```
`DefaultTitleStrategy` is a private default implementation.

SkyZeroZx replied:
Updated

SkyZeroZx commented on adev/src/content/guide/routing/define-routes.md:
```diff
@@ -299,6 +299,48 @@ const routes: Routes = [
+
+@Injectable({
+    providedIn: 'root',
+})
```
Done

JeanMeche commented on adev/src/content/guide/routing/define-routes.md:
```diff
@@ -299,6 +299,46 @@ const routes: Routes = [
+
+For advanced scenarios where you need centralized control over how the document title is composed, implement a `TitleStrategy`.
+
+`TitleStrategy` is an injectable abstract class that the router calls whenever it wants to set the page title for an activated route. The default implementation (`DefaultTitleStrategy`)
```
You should also probably, merge this line with the next paragraph

SkyZeroZx replied:
Done

JeanMeche commented on adev/src/content/guide/routing/define-routes.md:
````diff
@@ -299,6 +299,44 @@ const routes: Routes = [
+```ts
+import { provideRouter } from '@angular/router';
+import { AppTitleStrategy } from './app-title.strategy';
+
````
can we also add the import for `TitleStrategy`

JeanMeche replied:
My bad 🚀

JeanMeche commented on adev/src/content/guide/routing/define-routes.md:
```diff
@@ -299,6 +299,44 @@ const routes: Routes = [
 
 Route titles can also be set via a service extending the [`TitleStrategy`](/api/router/TitleStrategy) abstract class. By default, Angular uses the [`DefaultTitleStrategy`](/api/router/DefaultTitleStrategy).
 
+### Using TitleStrategy for page titles
```
While we're at it, we should update the @see link in `packages/router/src/page_title_strategy.ts`

SkyZeroZx replied:
Done, I hadn't taken that into account

SkyZeroZx committed:
docs: Adds guide for customizing page titles with TitleStrategy

Improves documentation by explaining how to implement and configure a custom title strategy for centralized control over page titles

JeanMeche reviewed (approved):
Thanks for this ! 🚀

eneajaho commented on adev/src/content/guide/routing/define-routes.md:
```diff
@@ -299,6 +299,44 @@ const routes: Routes = [
+  updateTitle(snapshot: RouterStateSnapshot): void {
+    // PageTitle is equal to the "Title" of a route if it's set
+    // If its not set it will use the "title" given in index.html
+    const pageTitle = this.buildTitle(snapshot) || this.title.getTitle();
```
Adding an example of how buildTitle would look like would be great. As this code will be copied over and over by devs.
